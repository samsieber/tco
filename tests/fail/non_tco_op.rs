#[tco::rewrite]
fn fac_with_acc(n: u128, acc: u128) -> u128 {
    if n > 1 {
        fac_with_acc(n - 1, acc * n) + 5
    } else {
        acc
    }
}

pub fn main(){
    fac_with_acc(5,1);
}