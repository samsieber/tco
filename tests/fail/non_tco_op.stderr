error: unreachable expression
 --> $DIR/non_tco_op.rs:4:40
  |
4 |         fac_with_acc(n - 1, acc * n) + 5
  |         ----------------------------   ^ unreachable expression
  |         |
  |         any code following this expression is unreachable
  |
note: the lint level is defined here
 --> $DIR/non_tco_op.rs:1:1
  |
1 | #[tco::rewrite]
  | ^^^^^^^^^^^^^^^
  = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)
